platform/security/handler/security-handler-pki/src/test/java/org/codice/ddf/security/handler/pki/CrlCheckerTest.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.handler.pki;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Properties;
+
+import org.jasypt.contrib.org.apache.commons.codec_1_3.binary.Base64;
+import org.junit.Test;
+
+public class CrlCheckerTest {
```

Comment:
The testEnabledCrlFileNoCertsGivenPasses() test covers the null case, I'll add one in for an empty array though. If the loadProperties fails to open the file, I should return an empty Properties object which causes it's getProperty to return null. This is treated the same way as the file existing but without the property set, which disables the CRL check and gives a bunch of "CRL is not set" errors and logs. I'll add in a unit test to make sure that is what actually happens though :)